tests/users/test_dao.py

```diff
@@ -148,6 +150,150 @@ def test_dao_delete_only_user_admin(self):
             (messages.USER_CANT_DELETE, HTTPStatus.BAD_REQUEST), dao_result
         )
 
+    def test_dao_delete_user(self):
+        dao = UserDAO()
+
+        user = UserModel(
+            name=user2["name"],
+            email=user2["email"],
+            username=user2["username"],
+            password=user2["password"],
+            terms_and_conditions_checked=user2["terms_and_conditions_checked"],
+        )
+        db.session.add(user)
+        db.session.commit()
+
+        # Verify that user was inserted in database through DAO
+        before_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNotNone(before_delete_user)
+
+        # Verify email
+        token = generate_confirmation_token(user2["email"])
+        result = dao.confirm_registration(token)
+        self.assertTrue(user.is_email_verified)
+
+        dao_result = dao.delete_user(2)
+
+        after_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNone(after_delete_user)
+        self.assertEqual(
+            (messages.USER_SUCCESSFULLY_DELETED, HTTPStatus.OK), dao_result
+        )
+
+    def test_dao_delete_user_that_does_not_exist(self):
+        dao = UserDAO()
+
+        # Verify that user does not exist
+        before_delete_user = UserModel.query.filter_by(id=2).first()
+        self.assertIsNone(before_delete_user)
+
+        dao_result = dao.delete_user(2)
+
+        after_delete_user = UserModel.query.filter_by(id=2).first()
```

Comment:
I don't think this is necessary since on line 188 above you've confirm that the user does not exist.

Reply:
And decorator also is used here:
<img width="547" alt="Screen Shot 2021-06-30 at 11 08 07 pm" src="https://user-images.githubusercontent.com/29667122/123968716-d4318a80-d9fa-11eb-85e7-4e363de3410e.png">

Reply:
> I don't think this is necessary since on line 188 above you've confirm that the user does not exist.

Yes, thanks. I'll remove it.